homeassistant/components/backup/manager.py

```diff
@@ -154,23 +167,14 @@ async def async_post_backup_actions(self, **kwargs: Any) -> None:
 
     async def load_platforms(self) -> None:
         """Load backup platforms."""
-        if self.loaded_platforms:
-            return
-        await integration_platform.async_process_integration_platforms(
-            self.hass,
-            DOMAIN,
-            self._add_platform_pre_post_handlers,
-            wait_for_platforms=True,
-        )
```

Comment:
`async_process_integration_platforms` can only be called once per domain, hence refactored so all platform registration happens in one callback